app/src/main/java/com/hedvig/app/feature/genericauth/otpinput/OtpInputActivity.kt

```diff
@@ -0,0 +1,117 @@
+package com.hedvig.app.feature.genericauth.otpinput
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.compose.setContent
+import androidx.compose.material.Scaffold
+import androidx.compose.material.rememberScaffoldState
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import com.google.accompanist.insets.systemBarsPadding
+import com.hedvig.app.BaseActivity
+import com.hedvig.app.R
+import com.hedvig.app.feature.loggedin.ui.LoggedInActivity
+import com.hedvig.app.ui.compose.composables.ErrorDialog
+import com.hedvig.app.ui.compose.composables.appbar.TopAppBarWithBack
+import com.hedvig.app.ui.compose.theme.HedvigTheme
+import com.hedvig.app.util.extensions.compatSetDecorFitsSystemWindows
+import com.hedvig.app.util.extensions.openEmail
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import org.koin.core.parameter.parametersOf
+
+class OtpInputActivity : BaseActivity() {
+
+    val model: OtpInputViewModel by viewModel {
+        parametersOf(
+            intent.getStringExtra(OTP_ID_EXTRA) ?: throw IllegalArgumentException(
+                "Programmer error: Missing OTP_ID in ${this.javaClass.name}"
+            ),
+            intent.getStringExtra(CREDENTIAL_EXTRA) ?: throw IllegalArgumentException(
+                "Programmer error: Missing CREDENTIAL in ${this.javaClass.name}"
+            )
+        )
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        window.compatSetDecorFitsSystemWindows(false)
+
+        setContent {
+            val scaffoldState = rememberScaffoldState()
+
+            HedvigTheme {
+                Scaffold(
+                    topBar = {
+                        TopAppBarWithBack(
+                            onClick = ::onBackPressed,
+                            title = stringResource(R.string.login_navigation_bar_center_element_title)
+                        )
+                    },
+                    scaffoldState = scaffoldState,
+                    modifier = Modifier.systemBarsPadding(top = true),
+                ) {
+                    val viewState by model.viewState.collectAsState()
+                    val events = model.eventsFlow.collectAsState(initial = null)
```

Comment:
Probably the same comment as above about saving events as Compose State<T>. Even though here you're launching a new LaunchedEffect when the value changes maybe we want to have one approach that we always take for event handling.